python/add_corona.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+#===================================================
+#
+#    Copyright (c) 2025
+#      SMASH Hybrid Team
+#
+#    GNU General Public License (GPLv3 or later)
+#
+#===================================================
+
+import argparse
+import numpy as np
+import sys
+import os
+import subprocess
+
+'''
+    This script adds the corona particles from the initial SMASH run
+    and possibly others to the sampled particle list.
+'''
+
+def find_out_event_number(filepath):
+    '''
+        Look into the last line of the file and find the number of events
+    '''
+    if not os.path.isfile(filepath):
+        print(filepath+" does not exist!", file=sys.stderr)
+        sys.exit(2)
+    tail = subprocess.run(['tail', '-n', '1', filepath],
+                          stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    last_line = tail.stdout.decode().strip().split()
+    n_events = 0
+    error = False
+    if "end" in last_line:
+        try:
+            n_events = int(last_line[2])
+        except ValueError:
+            error = True
+    else:
+        error = True
+    if error:
+        print("The number of events cannot be found for " + filepath,
+                file=sys.stderr)
+        sys.exit(3)
+    return n_events+1
+
+def extract_particles(filename):
+    '''
+        Extract particles from a file to a numpy array
+    '''
+    particles = np.empty(shape=[0, 13])
+    n_event = 0
+
+    for line in open(filename, "r"):
+        line = line.split()
+        # event end line
+        if "end" in line:
+            n_event += 1
+        # ignore header
+        elif "#" in line or "#!OSCAR2013" in line or "#!OSCAR2013Extended" in line:
+            continue
+        # particle line
+        else:
+            particle = np.array(line[:12])
+            particle = np.append(int(n_event), particle)
+            particles = np.append(particles,[particle], axis=0)
+    return(particles)
+
+def gather_corona_particles(corona_lists):
+    '''
+        Extract particle lines from the initial conditions oscar output of SMASH
+        and any extra file (e.g. from vHLLE oscar output) and join particles
+        into one array.
+    '''
+    is_any_corona_file_present = False
+    corona_particles = np.empty(shape=[0, 13])
+    for file in corona_lists:
+        if os.path.isfile(file):
+            is_any_corona_file_present = True
+            more_corona_particles = extract_particles(file)
+            corona_particles = np.append(corona_particles, more_corona_particles, axis=0)
+    if len(corona_particles) == 0:
+        print("No corona particles!", file=sys.stderr)
+        sys.exit(4)
+    if not is_any_corona_file_present:
+        print("Corona files not found!", file=sys.stderr)
+        sys.exit(2)
+    return corona_particles
+
+def copy_header(input_file, output_file):
+    header = ""
+    for line in open(input_file, "r"):
+        if line.startswith('# event'):
```

Comment:
I think this adds additional safety if the user toys around with `particle_lists.oscar` files in between running stages.
```suggestion
        if line.startswith('# event') or not line.startswith('#'):
```